provider/malicious_sources/resource_email_domain_alert.go

```diff
@@ -0,0 +1,194 @@
+				Description: "description of the policy",
+				Optional:    true,
+			},
+			"rule_action": {
+				Type:        schema.TypeString,
+				Description: "Need to provide the action to be performed ",
+				Optional:    true,
+				Default: "BLOCK",
+			},
+			"event_severity": {
+				Type:        schema.TypeString,
+				Description: "Generated event severity among LOW,MEDIUM,HIGH,CRITICAL",
+				Required:    true,
+			},
+			"environment": {
+				Type:        schema.TypeSet,
+				Description: "environment where it will be applied",
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"data_leaked_email": {
+				Type:        schema.TypeBool,
+				Description: "Users from leaked email domain",
+				Required:    true,
+			},
+			"disposable_email_domain": {
+				Type:        schema.TypeBool,
+				Description: "Users from disposable email domain",
+				Required:    true,
+			},
+			"email_domains": {
+				Type:        schema.TypeSet,
+				Description: "Email domains for rule",
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"email_regexes": {
+				Type:        schema.TypeSet,
+				Description: "Email regexes for rule",
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"email_fraud_score": {
+				Type:        schema.TypeString,
+				Description: "Minimum email fraud score",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceEmailDomainAlertCreate(d *schema.ResourceData, meta interface{}) error {
+	event_severity := d.Get("event_severity").(string)
+	name := d.Get("name").(string)
+	rule_action := d.Get("rule_action").(string)
+	data_leaked_email := d.Get("data_leaked_email").(bool)
+	email_fraud_score := d.Get("email_fraud_score").(string)
+	disposable_email_domain := d.Get("disposable_email_domain").(bool)
+	email_domains := d.Get("email_domains").(*schema.Set).List()
+	email_regexes := d.Get("email_regexes").(*schema.Set).List()
+	description := d.Get("description").(string)
+	environment := d.Get("environment").(*schema.Set).List()
+
+	envQuery := custom_signature.ReturnEnvScopedQuery(environment)
+	emailFraudScoreQuery := ReturnEmailFraudScoreQuery(email_fraud_score)
+	query := fmt.Sprintf(CREATE_EMAIL_DOMAIN_ALERT,name,description,event_severity,rule_action,data_leaked_email,disposable_email_domain,strings.Join(common.InterfaceToStringSlice(email_domains),","),strings.Join(common.InterfaceToStringSlice(email_regexes),","),emailFraudScoreQuery,envQuery)
+	var response map[string]interface{}
+	responseStr, err := common.CallExecuteQuery(query, meta)
+	log.Printf("This is the graphql query %s", query)
+	log.Printf("This is the graphql response %s", responseStr)
+	err = json.Unmarshal([]byte(responseStr), &response)
+	if err != nil {
+		fmt.Println("Error:", err)
+	}
+	id := response["data"].(map[string]interface{})["createMaliciousSourcesRule"].(map[string]interface{})["id"].(string)
```

Comment:
you're directly accessing map keys like response["data"] without checking if the key exists, which might cause a panic if the key is missing.
check if the key exists in the map before accessing it.

Reply:
done here https://github.com/Traceableai/terraform-provider-traceable/pull/49/commits/3316faf7ad1005247c027dc160b43e5a9eeb1542